lib/views/create-bug.js

```diff
@@ -80,48 +79,51 @@ function drawAttachments() {
 // API needs anyway, may want to avoid encoding until they are sent
 function inputChanged(e) {
   var files = e.target.files;
+  var attachments = [];
   for (var i = 0; i < files.length; i++) {
-    var file = files.item(i);
-    var reader = new FileReader();
-    reader.onload = (function(theFile) {
-      return function(e) {
-        bug.attachments.push({
-          name: theFile.name,
-          type: theFile.type,
-          data: e.target.result.split(',')[1]
-        });
-        bugChanged();
-        drawAttachments();
-      }
-    })(file);
-    reader.readAsDataURL(file);
+    attachments.push(pushAttachment(files.item(i)));
   }
+  Promise.all(attachments).then(bugChanged);
+}
+
+// Add the blobs coming from the capture logs activity
+// to the current bug.
+function addActivityAttachments(blobs, names) {
+  Promise.all(blobs.map(function(blob, i) {
+    return pushAttachment(blob, names[i]);
+  })).then(bugChanged);
 }
 
 
-function addAttachments(blobs, names) {
-  blobs.forEach(function(blob, i) {
-    var reader = new window.FileReader();
-    reader.readAsDataURL(blob);
-    reader.onloadend = function() {
+function pushAttachment(file, name) {
+  return new Promise(function(resolve) {
+    var reader = new FileReader();
+    reader.onload = function(e) {
       bug.attachments.push({
-        name: names[i],
-        type: blob.type,
-        data: reader.result
+        name: name || file.name,
+        type: file.type,
+        data: e.target.result.split(',')[1]
       });
-      bugChanged();
-      drawAttachments();
-    }
+      resolve();
+    };
+    reader.onerror = reader.onabort = function(error) {
```

Comment:
You'll find that once you want to start handling issues like this, having the ```reject()``` calls already in place is very valuable.